src/magic.cpp

```diff
@@ -3465,8 +3470,10 @@ int cast_sanctuary( ubyte level, CHAR_DATA *ch, char *arg, int type,
 int spell_camouflague(ubyte level, CHAR_DATA *ch, CHAR_DATA *victim, struct obj_data *obj, int skill) 
 {
   struct affected_type af;
-
-  if(!affected_by_spell(victim, SPELL_CAMOUFLAGE)) {
+   if(affected_by_spell(victim, SPELL_CAMOUFLAGE)) {
+     send_to_char("You are already well hidden enough!\n\r", ch);
+     return eFAILURE;
+   }
      act("$n fades into the plant life.", victim, 0, 0, TO_ROOM, INVIS_NULL);
```

Comment:
Line indentation on the rest of this needs to be fixed as it's no longer inside the block.   This is a problem in a lot of your functions, so I'm just listing it once here.